pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -36,8 +37,54 @@ def __init__(self, runtime: Runtime, version: str, assembly: str, publish: bool,
         self.logger = runtime.logger
         self.working_dir = self.runtime.working_dir
         self.fail_count_name = f'{version}-assembly.{assembly}.count'
+        self.job_run = self.runtime.get_job_run_url()
+
+    async def comment_on_assembly_pr(self):
+        """
+        Comment the link to this jenkins build on the assembly PR if it was triggered automatically
+        """
+        owner = "openshift-eng"
+        repository = "ocp-build-data"
+        branch = self.doozer_data_gitref
+        token = os.environ.get('GITHUB_TOKEN')
+
+        api = GhApi(owner=owner, repo=repository, token=token)
+
+        # Head needs to have the repo name prepended for GhApi to fetch the correct one
+        head = f"{owner}:{branch}"
+        # Find our assembly PR.
+        prs = api.pulls.list(head=head, state="open")
+
+        if len(prs) != 1:
+            self.logger.error(f"{len(prs)} PR(s) were found from the auto generated branch {branch}. We need only 1.")
+            return
+
+        if len(prs) == 0:
+            self.logger.error("No assembly PRs were found from branches with naming convention eg: auto-gen-assembly-openshift-4.11-4.11.47")
```

Comment:
```suggestion
            self.logger.error(f"No assembly PRs were found with head={head}")
```

Reply:
build-sync job doesn't care about the branch name, the branch name can be anything
and it will fetch and run